PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -1219,11 +1249,15 @@ PayloadMain (
       BootShell    = TRUE;
     }
   } else {
-    if (OsBootOptionList->BootToShell != 0) {
-      BootShell    = TRUE;
-    } else if (DebugCodeEnabled ()) {
-      ShellTimeout = (UINTN)PcdGet16 (PcdPlatformBootTimeOut);
-      BootShell    = TRUE;
+    if (OsBootOptionList->RestrictedBoot != 0) {
+      BootShell    = FALSE;
+    } else {
+      if (OsBootOptionList->BootToShell != 0) {
+        BootShell    = TRUE;
+      } else if (DebugCodeEnabled ()) {
+        ShellTimeout = (UINTN)PcdGet16 (PcdPlatformBootTimeOut);
+        BootShell    = TRUE;
+      }

```

Comment:
If we boot to shell firstly, then boot to setup, is there any impact?
If it works, the setup boot option is same with other boot options, maybe we don't need add this limitation. and this behavior is also aligned with UEFI bios. (they could boot to shell firstly, then entering setup menu).

Reply:
I think we can keep this capability.  But it is platform's choice to determine if it needs to set this flag.  
For production that needs this,  I would think the restriction is required for security reason.